org.eclipse.wildwebdeveloper.xml/src/org/eclipse/wildwebdeveloper/xml/internal/ui/preferences/XMLCatalogPreferencePage.java

```diff
@@ -0,0 +1,279 @@
+		viewer.setContentProvider(new EntriesContentProvider());
+		viewer.setLabelProvider(new EntriesLabelProvider());
+		entries = new TreeSet<String>(String.CASE_INSENSITIVE_ORDER);
+		
+		for (String filepath : store.getString(XMLPreferenceInitializer.XML_PREFERENCES_CATAGLOGS).split(",")) {
+			if (!filepath.isEmpty()) {
+				entries.add(filepath);
+			}
+		}
+
+		viewer.setInput(entries);
+		viewer.getControl().setLayoutData(new GridData(GridData.FILL_BOTH));
+		viewer.expandAll();
+
+		Composite buttonComposite = new Composite(entriesGroup, SWT.NONE);
+		buttonComposite.setLayout(new GridLayout(1, false));
+		buttonComposite.setLayoutData(new GridData(SWT.FILL, SWT.TOP, false, false));
+
+		Button addButton = new Button(buttonComposite, SWT.PUSH);
+		addButton.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+		addButton.setText(Messages.PreferencesPage_Add);
+		addButton.addSelectionListener(new SelectionListener() {
+
+			public void widgetDefaultSelected(SelectionEvent e) {
+			}
+
+			public void widgetSelected(SelectionEvent e) {
+				String result = openSelectFileDialog(null);
+				if (result == null || result.trim().length() == 0) {
+					return;
+				}
+				String newEntry = new File(result).getAbsolutePath();
+				entries.add(newEntry);
+				viewer.refresh();
+				viewer.setSelection(new TreeSelection(new TreePath(new String[] {newEntry})));
+			}
+		});
+		final Button changeButton = new Button(buttonComposite, SWT.PUSH);
+		changeButton.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+		changeButton.setText(Messages.PreferencesPage_Change);
+		changeButton.setEnabled(false);
+		changeButton.addSelectionListener(new SelectionListener() {
+
+			public void widgetSelected(SelectionEvent e) {
+				if (selectedEntry == null) {
+					return;
+				}
+
+				String result = openSelectFileDialog(selectedEntry);
+				if (result == null || result.trim().length() == 0) {
+					return;
+				}
+				String newEntry = new File(result).getAbsolutePath();
+				entries.remove(selectedEntry);
+				entries.add(newEntry);
+				viewer.refresh();
+				viewer.setSelection(new TreeSelection(new TreePath(new String[] {newEntry})));
+
+			}
+
+			public void widgetDefaultSelected(SelectionEvent e) {
+
+			}
+		});
+		final Button removeButton = new Button(buttonComposite, SWT.PUSH);
+		removeButton.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+		removeButton.setText(Messages.PreferencesPage_Remove);
+		removeButton.setEnabled(false);
+
+		removeButton.addSelectionListener(new SelectionListener() {
+
+			public void widgetSelected(SelectionEvent e) {
+				if (selectedEntry != null) {
+					entries.remove(selectedEntry);
+					viewer.refresh();
+				}
+			}
+
+			public void widgetDefaultSelected(SelectionEvent e) {
+			}
+		});
+
+		viewer.addSelectionChangedListener(new ISelectionChangedListener() {
+
+			public void selectionChanged(SelectionChangedEvent event) {
+				changeButton.setEnabled(false);
```

Comment:
I don't think you need to care about the old selection here, you can use directly `event.getSelection()` with 
```
changeButton.setEnabled(!event.getSelection().isEmpty());
editButton.setEnabled(!event.getSelection().isEmpty());
```

Reply:
Old selection is used to allow a user to 'select' and 'deselect' a catalog file. But probably it has no practical sense here.